core/src/mender-api.c

```diff
@@ -272,7 +278,22 @@ api_check_for_deployment_v2(int *status, void *response) {
         goto END;
     }
 
-    /* TODO: Retrieve artifact name from store (see ticket MEN-7479) */
+#ifdef CONFIG_MENDER_PROVIDES_DEPENDS
+#ifdef CONFIG_MENDER_FULL_PARSE_ARTIFACT
+    /* Add provides from storage */
+    mender_key_value_list_t *provides = NULL;
+    if (MENDER_FAIL == mender_storage_get_provides(&provides)) {
+        mender_log_error("Unable to get provides");
+        goto END;
+    }
+    for (mender_key_value_list_t *item = provides; NULL != item; item = item->next) {
+        if (NULL == cJSON_AddStringToObject(json_provides, item->key, item->value)) {
+            mender_log_error("Unable to allocate memory");
+            goto END;
+        }
+    }
```

Comment:
We should destroy the list somewhere, right?

Reply:
Yes, you're right. Overlooked it